src/providers/queries/postQueries/wavesQueries.ts

```diff
@@ -355,3 +355,29 @@ export const fetchLatestWavesContainer = async (host) => {
 
   return _latestPost;
 };
+
+export const useDeleteWaveMutation = () => {
+  const queryClient = useQueryClient();
+
+  const _mutationFn = async (wavePermlink: any) => {
+    if (wavePermlink) {
+      return wavePermlink;
+    }
+
+    throw new Error('invalid mutations data');
+  };
+
+  const _options: UseMutationOptions<string, unknown, string, void> = {
+    onMutate: async (filteredWavesData: any) => {
+      await queryClient.cancelQueries(QUERIES.WAVES.GET);
+      queryClient.setQueryData([QUERIES.WAVES.GET], [...filteredWavesData]);
```

Comment:
@aliseyalvi this might cause issues or it might not do anything at all considering the accumulating wave keys and data is done in a relatively complicated way, check this below snippet for instance.

```
  const wavesQueries = useQueries({
    queries: activePermlinks.map((pagePermlink, index) => ({
      queryKey: [QUERIES.WAVES.GET, host, pagePermlink, index], // index at end is used to track query hydration
      queryFn: () => _fetchWaves(pagePermlink),
      initialData: [],
    })),
  });
```
 
A waves query key is a very combination of base key, host for waves, active container permlink and it's index in list of available containers. we can arrange a short meeting on how it all comes together
  
 Also the effects of cancelQueries also need assessing, for instance what would happen if it is invoked while new containers or waves were being fetched as a result of onEndReached call, will it affect user experience. might be a rare instance though.